src/com/dogonfire/exams/ExamManager.java

```diff
@@ -417,6 +421,21 @@ public void calculateExamResult(String playerName)
 
 			plugin.sendMessage(playerName, ChatColor.GREEN + "Congratulations, you passed the exam!");
 			plugin.sendToAll(ChatColor.GREEN + playerName + " just PASSED the " + ChatColor.YELLOW + plugin.getStudentManager().getExamForStudent(playerName) + ChatColor.GREEN + " exam!");
+			
+			// giving player an advancement for passing an exam if set to true
+			if (plugin.grantAdvancement) {
+				Advancement a = Bukkit.getAdvancement(NamespacedKey.fromString(plugin.examAdvacementName));
```

Comment:
Could be changed to see if the exam has an Advancement specified.

Reply:
agree

Reply:
For now one advancement works as a prof of concept. 
The reason you can turn this feature on and of is exactly because people might not have a datapack on their server or want to have one. And it does not have to be a specific datapack. Thats why it can be change in the config.yml so people can create their own datapacks with the advancement they what to give.